### PR TITLE
Git_Basics.rst: Incorporate http://ohshitgit.com/

### PR DESCRIPTION
**Pull request by yzgyyang**
http://ohshitgit.com/ contains helpful git snippets when something has
gone wrong.
This commit adds the link to the website as well as some snippets and
paraphrased instructions into Git Basics.

Closes https://github.com/coala/coala/issues/4985

### Checklist

- [x] I read the [commit guidelines](http://coala.io/commit) and I've followed
      them.
- [x] I ran coala over my code locally. (*All commits have to pass
      individually.* It is not sufficient to have "fixup commits" on your PR,
      our bot will still report the issues for the previous commit.) You will
      likely receive a lot of bot comments and build failures if coala does not
      pass on every single commit!

### CODE REVIEW

**Comment by yzgyyang** [comment]
I realized that we cannot just rephrase the contents since the site is not open-sourced with a license. I will try just the link and a short description instead. In the meantime, how do you suggest we do to the existing "Common Git Issues"? IMO it's a bit wordy and somewhat duplicate on what's on ohshitgit.

**Comment by jayvdb** [comment]
Ya, dont duplicate help which is on that website.
Our git help page should be specific information which we feel is required to help git newbies do their first PR with coala. That is beyond the scope of your issue.

**Comment by jayvdb** [comment]
ack https://github.com/coala/coala/pull/5387/commits/7f731f8f5e51f5380197333d8281e55af55e8655

**Comment by jayvdb** [comment]
@gitmate-bot ff

**Comment by yzgyyang** [comment]
@jayvdb It's weird that the related issue was not closed automagically. Should I do it?

**Comment by jayvdb** [comment]
https://github.com/coala/coala/issues/4985 was closed automatically.

**Comment by yzgyyang** [comment]
@jayvdb Sorry for the spam. I just refreshed the page and realized that. :P
